src/contracts/atlas/GasAccounting.sol

```diff
@@ -31,14 +32,22 @@ abstract contract GasAccounting is SafetyLocks {
     }
 
     /// @notice Borrows ETH from the contract, transferring the specified amount to the caller if available.
-    /// @dev Borrowing is only available until the end of the SolverOperations phase, for solver protection.
+    /// @dev Borrowing is only available until the end of the SolverOperation phase, for solver protection.
     /// @param amount The amount of ETH to borrow.
     function borrow(uint256 amount) external payable {
+        if (amount == 0) return;
+
+        // borrow() can only be called by the Execution Environment (by delegatecalling a DAppControl hook), and only
+        // during or before the SolverOperation phase.
         Lock memory _lock = lock;
         if (_lock.activeEnvironment != msg.sender) {
             revert InvalidExecutionEnvironment(_lock.activeEnvironment);
         }
-        if (_lock.phase > uint8(ExecutionPhase.SolverOperations)) revert WrongPhase();
+        if (_lock.phase > uint8(ExecutionPhase.SolverOperation)) revert WrongPhase();
+
+        // borrow() will revert if called after solver calls reconcile()
+        (, bool calledBack, bool fulfilled) = _solverLockData();
+        if (calledBack || fulfilled) revert WrongPhase();
```

Comment:
```suggestion
        (, bool calledBack, ) = _solverLockData();
        if (calledBack) revert WrongPhase();
```
Think this can be simplified by only checking `calledBack`, because there's no way to set `fulfilled` to `true` without calling `reconcile()` and also setting `calledBack` to `true`

Reply:
Fixed here https://github.com/FastLane-Labs/atlas/pull/271/commits/303ba5d526a541384dafdf1d178ef2e09bb24701